views.py

```diff
@@ -0,0 +1,79 @@
+from django.shortcuts import render, render_to_response, get_object_or_404, redirect
+from django.http import HttpResponseRedirect, HttpResponse, JsonResponse
+from django.template.context_processors import csrf
+from django.template import RequestContext
+import json
+import requests
+import urllib.parse
+from collections import defaultdict
+import datetime
+import dateutil.parser
+import json
+
+def home(request):
+	if request.method == 'POST' and request.is_ajax():
+		username = request.POST['username']
+		year = 2016 #datetime.datetime.now().year
+		# Starting date of the year
+		start_date = datetime.datetime(year, 1, 1, 00, 00, 00)
+		# Ending date of the year
+		end_date = datetime.datetime(year, 12, 31, 23, 59, 59)
+
+		generalParameters = {'action': 'query',
+		           'format': 'json',
+		           'list': 'usercontribs',
+		           'uclimit':'max',
+		           'ucuser':username,
+		           'ucdir':'newer',
+		           'ucnamespace':0,
+		           'ucstart':start_date.isoformat(' '),
+		           'ucend':end_date.isoformat(' ')
+		           }
+		articlesCreatedParameters = generalParameters.copy()
+		articlesEditedParameters = generalParameters.copy()
+		articlesCreatedParameters['ucshow'] = "new"
+		articlesEditedParameters['ucshow'] = "!new"
+		# For articles created
+		articlesCreatedDates = getContributionTimelineData(articlesCreatedParameters)
+		# For articles edited
+		articlesEditedDates = getContributionTimelineData(articlesEditedParameters)
+		return HttpResponse(json.dumps({'articlesCreatedDates':articlesCreatedDates, 'articlesEditedDates':articlesEditedDates}), content_type="application/json")
+	return render(request, 'home.html', {})
+
+
+def getContributionTimelineData(parameters):
+	contributionDatesDict = defaultdict(int) # default value of int is 0
+	url='https://en.wikipedia.org/w/api.php'
+	while True:
+		url_parameters = urllib.parse.urlencode(parameters)
+		formatted_url =  (url + '?{}').format(url_parameters)
+		response_object = requests.get(formatted_url)
+		if response_object.status_code == 200:
+				# Loading the response data into a dict variable
+				# json.loads takes in only binary or string variables so using text to fetch binary content
+				# Loads (Load String) takes a Json file and converts into python data structure (dict or list, depending on JSON)
+				json_data = json.loads(response_object.text)
+				try:
+					 # if username is not valid
+					if json_data['error']:
+						print("Error", json_data['error']['info'])
```

Comment:
This won't work with an `application/json` response type.

Reply:
Okay, now I am surfacing the error from Django to the Ajax call.